src/mds/Server.cc

```diff
@@ -6976,6 +6997,11 @@ void Server::handle_client_rename(MDRequestRef& mdr)
     respond_to_request(mdr, -EINVAL);
     return;
   }
+  if (is_operation_on_dot_or_dotdot(srcpath)) {
+    respond_to_request(mdr, -EBUSY);
+    return;
+  }
```

Comment:
destpath?

Reply:
for destpath, the file would be moved to its parent directory.

Reply:
We need to check rename to the current directory itself and return -EBUSY (`..` is valid).

Currently -ENOENT is returned.

Reply:
It does not make sense. Command 'rename  a ..' works, but the actually syscall is rename("a",  "../a")

Reply:
ok -- I had the impression that ceph_rename(to, "..") would send the rename request to the mds with the appropriate "parent-inode/name" combination.

So, srcpath and destpath should be caught here, correct?

Reply:
yes

Reply:
yes, checking both srcpath and destpath here should be enough